docs/sql-ref-syntax-qry-select-groupby.md

```diff
@@ -42,42 +42,60 @@ aggregate_name ( [ DISTINCT ] expression [ , ... ] ) [ FILTER ( WHERE boolean_ex
 
 ### Parameters
 
+* **grouping_expression**
+
+    Specifies the criteria based on which the rows are grouped together. The grouping of rows is performed based on
+    result values of the grouping expressions. A grouping expression may be a column name like `GROUP BY a`, a column position like
+    `GROUP BY 0`, or an expression like `GROUP BY a + b`.
+
 * **GROUPING SETS**
 
-    Groups the rows for each subset of the expressions specified in the grouping sets. For example,
-    `GROUP BY GROUPING SETS (warehouse, product)` is semantically equivalent
+    Groups the rows for each grouping set specified after GROUPING SETS. For example,
+    `GROUP BY GROUPING SETS ((warehouse), (product))` is semantically equivalent
     to union of results of `GROUP BY warehouse` and `GROUP BY product`. This clause
     is a shorthand for a `UNION ALL` where each leg of the `UNION ALL`
-    operator performs aggregation of subset of the columns specified in the `GROUPING SETS` clause.
-
-* **grouping_set**
-
-    A grouping set is specified by zero or more comma-separated expressions in parentheses.
-
-    **Syntax:** `( [ expression [ , ... ] ] )`
-
-* **grouping_expression**
-
-    Specifies the criteria based on which the rows are grouped together. The grouping of rows is performed based on
-    result values of the grouping expressions. A grouping expression may be a column alias, a column position
-    or an expression.
+    operator performs aggregation of each grouping set specified in the `GROUPING SETS` clause.
+    Similarly, `GROUP BY GROUPING SETS ((warehouse, product), (product), ())` is semantically
+    equivalent to the union of results of `GROUP BY warehouse, product`, `GROUP BY product`
+    and global aggregate.
+    
+    GROUPING SETS can be followed with normal expressions as well, such as
+    `GROUP BY GROUPING SETS (warehouse, product)`. It generates one grouping set
+    per non-empty subset of the given expressions, so `GROUP BY GROUPING SETS (warehouse, product)`
```

Comment:
I think we don't need this paragraph anymore. We should put `grouping_set` before this one so that people know `a` is equal to `(a)`.

Reply:
> I think we don't need this paragraph anymore. We should put `grouping_set` before this one so that people know `a` is equal to `(a)`.

Done